Register, Nihal_04.21.25

nihalerdal opened the pull request:
Added a registration modal with name, email, and password inputs. Registration sends data to the API and navigates to /bookmark on success.

Note: Input validation and dynamic navbar updates will be handled in a separate PR.

nihalerdal committed:
Create UI

nihalerdal committed:
Fix divider text color and checkbox font size

nihalerdal committed:
Add textfield styling into theme

nihalerdal committed:
Add functionality

nihalerdal committed:
Navigate next page after login

nihalerdal committed:
Merge with main

nihalerdal committed:
Code format

rezahedi commented on src/Components/Auth/Register.jsx:
```diff
@@ -0,0 +1,256 @@
+                  margin="normal"
+                  value={password}
+                  onChange={handlePasswordChange}
+                />
```
It's better to make the inputs required. Because when I hit Sign-in button without filling the form, instead of showing a message that I need to fill the form, it redirect me to bookmark page anyway.

rezahedi commented on src/Components/Auth/Register.jsx:
```diff
@@ -0,0 +1,256 @@
+
+    try {
+      const data = await postData(URL, requestBody);
+      console.log(data);
```
Suggestion: Consider saving the user's data (e.g., id, name, token, etc.) into localStorage after a successful registration. This can be useful for maintaining login state across pages like the dashboard or bookmark pages. You could do something like:
`localStorage.setItem("profile", JSON.stringify(data))`
This will help in identifying whether the user is logged in and make it easier to access user info throughout the app.

nihalerdal replied:
These are not included in this task yet, it'll be in the next one

rezahedi commented on src/Components/Auth/Register.jsx:
```diff
@@ -0,0 +1,256 @@
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleClose}
+      maxWidth={false}
+      slotProps={{
+        sx: {
+          width: { xs: "100%", sm: "90%", md: "80%" },
+          maxWidth: 600,
+          borderRadius: 3,
+        },
+      }}
+    >
```
Cool feature, I really liked that the register component pops up as a dialog box, good one @nihalerdal

rezahedi commented on src/util/index.js:
```diff
@@ -22,4 +22,19 @@ const getAllData = async (url) => {
+    const res = await axios.post(url, requestBody, config);
+    return res.data;
+  } catch (error) {
+    console.error("POST error in", url, error);
```
Suggestion: It would be great to display the error message returned from the API somewhere below or above the `Sign in` button, so the user can understand what went wrong. The API returns a 400 status code along with a response like:
`{ msg: "<the message describing why it's not working>" }`
Showing this message in the UI will improve the user experience by providing clear feedback on registration or login issues.

rezahedi reviewed (commented):
The UI looks great and slick, Thanks @nihalerdal

conquerorabhishek commented on src/util/functions.js:
```diff
@@ -53,4 +53,21 @@
 // // getPlansByCategory("academic-and-cultural-tour");
 
-// export { getAllPlans, getPlansByCategory, getSinglePlan, getPlansByUserId };
+// --------------------------------------``** USER REGISTER **``  --------------------------------------
```
nit: Why it's commented? The better practice if code is not needed then don't keep it. we can always bring the code from history if needed.

nihalerdal committed:
Manage user login status , store user info and update navbar accordingly

nihalerdal committed:
Set required for input

nihalerdal committed:
Update format

nihalerdal commented:
Changes included:

- Implemented user login status management.
- Stored user information (name, email, image, token) upon successful login.
- Updated the navbar to show personalized information (hello message and emoji) after login.
- Added a logout button to allow users to log out.

rezahedi reviewed (approved):
@nihalerdal Great work on the auth provider! Just a couple of things: client-side validation still needs improvement, right now, hitting "Sign In" with empty fields or invalid email or password shows no feedback. Also, no error message if the API fails. If that’s coming in the next task, then everything else looks good to me.

StuvaScript reviewed (commented):
Great design! I wasn't able to actually create an account tho. I kept getting a 400 bad request error. Not sure what was happening...

nihalerdal committed:
Fix unique key and proptype issues

nihalerdal committed:
Resolve merge conflict
